codegen/src/main/java/software/amazon/awssdk/codegen/emitters/tasks/PackageInfoGeneratorTasks.java

```diff
@@ -38,17 +40,100 @@ public final class PackageInfoGeneratorTasks extends BaseGeneratorTasks {
     @Override
     protected List<GeneratorTask> createTasks() throws Exception {
         Metadata metadata = model.getMetadata();
-        String packageInfoContents =
-            String.format("/**%n"
-                          + " * %s%n"
-                          + "*/%n"
-                          + "package %s;",
-                          metadata.getDocumentation(),
-                          metadata.getFullClientPackageName());
+
+        String baseDocumentation = metadata.getDocumentation();
+
+        String codeExamples = getCodeExamples(metadata);
+        
+        StringBuilder sb = new StringBuilder();
+        sb.append(String.format("/**%n * %s%n", baseDocumentation));
+        if (!codeExamples.isEmpty()) {
+            sb.append(String.format(" *%n * %s%n", codeExamples));
+        }
+        sb.append(String.format("*/%npackage %s;", metadata.getFullClientPackageName()));
+        String packageInfoContents = sb.toString();
         return Collections.singletonList(new SimpleGeneratorTask(baseDirectory,
                                                                  "package-info.java",
                                                                  model.getFileHeader(),
                                                                  () -> packageInfoContents));
     }
 
+    String getCodeExamples(Metadata metadata) {
+        String exampleMetaPath = "software/amazon/awssdk/codegen/example-meta.json";
+        List<DocumentationUtils.ExampleData> examples = 
+            DocumentationUtils.getServiceCodeExamples(metadata, exampleMetaPath);
+        
+        if (examples.isEmpty()) {
+            return "";
+        }
+        
+        String codeExamplesJavadoc = generateCodeExamplesJavadoc(examples);
+
+        StringBuilder result = new StringBuilder();
+        String[] lines = codeExamplesJavadoc.split("\n");
+        for (String line : lines) {
+            if (!line.trim().isEmpty()) {
+                result.append(line);
+                if (!line.equals(lines[lines.length - 1])) {
+                    result.append(System.lineSeparator()).append(" * ");
+                }
+            }
+        }
+        
+        return result.toString();
+    }
+    
+
+    private String generateCodeExamplesJavadoc(List<DocumentationUtils.ExampleData> examples) {
+        Map<String, List<DocumentationUtils.ExampleData>> categorizedExamples = new java.util.LinkedHashMap<>();
```

Comment:
Let's replace qualified name with import. `new java.util.LinkedHashMap<>();` -> `new LinkedHashMap<>())`

Can be shortened to the following:

```java
        Map<String, List<DocumentationUtils.ExampleData>> categorizedExamples = 
            examples.stream().collect(Collectors.groupingBy(DocumentationUtils.ExampleData::getCategory, 
                                                            LinkedHashMap::new, 
                                                            Collectors.toList()));

```